packages/components/src/DateTimePickers/pickers/YearPicker/YearPicker.test.js

```diff
@@ -4,11 +4,89 @@ import { shallow } from 'enzyme';
 import YearPicker from './YearPicker.component';
 
 describe('YearPicker', () => {
-	it('should render a YearPicker', () => {
-		// when
-		const wrapper = shallow(<YearPicker />);
+	it('should render', () => {
+		const wrapper = shallow(<YearPicker
+			selectedYear={2012}
+		/>);
 
-		// then
 		expect(wrapper.getElement()).toMatchSnapshot();
 	});
+
+	it('should have exactly one selected year', () => {
+		const wrapper = shallow(<YearPicker
+			selectedYear={2012}
+		/>);
+
+		const actions = wrapper.find('PickerAction');
+
+		expect(actions.length).toBe(5);
+
+		const selectedActions = actions.filterWhere(action => action.prop('isSelected') === true);
+
+		expect(selectedActions.length).toBe(1);
+
+		selectedActions.forEach(action => {
+			expect(action.prop('label')).toBe('2012');
+		});
+	});
+
+	it('should callback with the year picked', () => {
+		const yearToSelect = 2013;
+		const onSelect = jest.fn();
+
+		const wrapper = shallow(<YearPicker
+			selectedYear={2012}
+			onSelect={onSelect}
+		/>);
+
+		const nextYearAction = wrapper
+			.find('PickerAction')
+			.findWhere(n => n.prop('label') === yearToSelect.toString());
+
+		expect(nextYearAction.length).toBe(1);
```

Comment:
You don't need that. It is implicitly tested with the snapshot of the first test

Reply:
✔